src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/DateTimeMarshaler.cs

```diff
@@ -0,0 +1,42 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+
+namespace System.Runtime.InteropServices.JavaScript
+{
+    public static class DateTimeMarshaler
+    {
+        public static string FromJavaScriptPreFilter => "return BINDING._pre_filter_date(value)";
```

Comment:
`Transformer` or `Transform` could be better name or possibly `Expression`. I would drop `Pre` and `Post` from the name.
So we could have `FromJavascriptExpression`

Reply:
I changed them to ```JavaScriptToInterchangeTransform``` and ```InterchangeToJavaScriptTransform``` which I believe is clear and explicit - the representation being converted to is an 'interchange format' that can cross the js/wasm boundary in both directions. The C# side is still FromJavaScript/ToJavaScript but that's because an interchange format is not always being used. It's a 'transform' because it's not a single-statement expression (I'm open to removing support for multiple statements later, but right now it would make tests and benchmarks harder to write and wouldn't simplify the JS that much)

I think this addresses these concerns, feel free to unresolve if you disagree.